api/firmware/attestation.go

```diff
@@ -27,14 +27,40 @@ import (
 	"github.com/digitalbitbox/bitbox02-api-go/util/semver"
 )
 
+type attestationPubkey struct {
+	// pubKeyHex is the uncompressed SECP256k1 attestation pubkey, hex-encoded.
+	pubkeyHex string
+	// acceptedBootloaderHashHex, if non-nil, is a hex-encoded bootloader hashes (of the padded
+	// bootloader binary, i.e. the device bootloader area), for which this attestation pubkey is
+	// valid. If nil, any bootloader will be accepted.
+	acceptedBootloaderHashHex *string
+}
+
+func ref(s string) *string {
+	return &s
+}
+
 // attestationPubkeys is a map of attestation pubkey identifier to attestation pubkey.
-// the pubkey is an uncompressed SECP256k1 pubkey.
-var attestationPubkeys = map[string]string{
-	"f36581299c784acfe26d735c1f937f7e397ee471a02983906a29660f3eee2004": "04074ff1273b36c24e80fe3d59e0e897a81732d3f8e9cd07e17e9fc06319cd16b25cf74255674477b3ac9cbac2d12f0dc27a662681fcbc12955b0bccdcbbdcfd01",
-	"4c2b7ec9399038da906633173db24f017cdba1a63fe2a84548319701a0b42039": "044c53a84f41fa7301b378bb3c260fc9b2ff1cbea7a78181279a8566797a736f12cea25fa2b1c27a844392fe9b37547dc6fbd00a2676b816e7d2d3562be2a0cbbd",
-	"f24b6ffaae3cd4905ab07734afeb4d179a6b6954d28e710c98ddab9a53b91f44": "04e9c8dc929796aac65af5084eb54dc1ee482d5e0b5c58e2c93f243c5b70b21523324bdb78d7395317da165ef1138826c3ca3c91ca95e6f490c340cf5508a4a3ec",
-	"0d39866fb5b0f81cdd559447ef3378a1f3e3e45e40b307a5fcbc5e02c67a967c": "04c2fb05889b9dff5a9fb22a59ee1d16bfc2863f0400ddcb69566e2abe8a15fa0ba1240254ca45aa310d170e724e1310ce5f611cada76c12e3c24a926a390ca4be",
-	"10cecd28b7ed38fd08406728d149762007abd694a358f06448d3ee1dcd9d908a": "04c4e82d6d1b91e7853eba96a871ad31fc62620b826b0b8acf815c03de31b792a98e05bb34d3b9e0df1040eac485f03ff8bbbf7a857ef1cf2a49a60ac084efb88f",
+var attestationPubkeys = map[string]attestationPubkey{
```

Comment:
Or maybe I would consider going even further: move here what's in performAttestation. For example:

 ```go
// attestationKey returns a root pubkey suitable to perform device attestation.
// The id argument is the key identifier obtained from the device using the "attestation" op.
// The boothash is compared to an expected value of the bootloader binary hash associated with the given identifier.
func attestationKey(id string, boothash []byte) (*ecdsa.PublicKey, error) {
  var hexkey, goodhash string
  switch id {
  // bb02 firmware x
  case "f36581299c784acfe26d735c1f937f7e397ee471a02983906a29660f3eee2004":
      hexkey = "04074ff1273b..."
  // bb base something
  case "....":
      hexkey = "abcd"
      goodhash = "fdea"
  default:
      return nil, errBadPubkeyIdentifier
  }
  if goodhash != "" && goodhash != hex.EncodeToString(boothash) {
      return nil, errBadBootloaderHash
  }
  b, err := hex.DecodeString(hexkey)
  if err != nil {
      return nil, err
  }
  key, err := btcec.ParsePubKey(b, btcec.S256())
  if err != nil {
      return nil, err
  }
  return key.ToECDSA(), nil
}
```

Then in performAttestation it would look like:

```go
rootPubkey, err := attestationKey(hex.EncodeToString(rootPubkeyIdentifier), bootloaderHash)
if err != nil {
  return false, err
}
// ...
if !verify(rootPubkey, certMsg.Bytes(), certificate) {
  return false, nil
}
```

Reply:
Good ideas, but I think the risk is small with the unexported map, and it makes it slightly easier to add a mock entry in the test.